models/review_model.js

```diff
@@ -39,18 +40,49 @@ exports.updateReviews = (review_id, inc_votes) => {
     });
 };
 
-exports.fetchAllReviews = () => {
-  return db
-    .query(
-      `SELECT reviews. *, COUNT(comments) AS comment_count
-      FROM reviews
-      LEFT JOIN comments
-      ON reviews.review_id = comments.review_id
-      GROUP BY reviews.review_id
-      ORDER BY reviews.created_at DESC
-      `
-    )
-    .then((results) => {
-      return results.rows;
-    });
+exports.fetchAllReviews = (
+  sort_by = "created_at",
+  order = "desc",
+  category
+) => {
+  const validSortBy = [
+    "owner",
+    "title",
+    "review_id",
+    "category",
+    "created_at",
+    "votes",
+    "comment_count",
+  ];
+  const categoryArray = []
+
+  let queryStr = `SELECT reviews. *, COUNT(comments) AS comment_count
+  FROM reviews
+  LEFT JOIN comments
+  ON reviews.review_id = comments.review_id`;
+
+  if (category) {
+    queryStr += ` WHERE category = $1 GROUP BY reviews.review_id`;
+    categoryArray.push(category)
+  } else {
```

Comment:
I don't think you want an else here. Write a test which filters by category which has some comments. You should see it failing